docs/README.md

```diff
@@ -45,3 +46,104 @@ It boasts a number of handy features:
   - inspection of blocks, transactions, results and events
 
 [GitHub repository]: https://github.com/oasisprotocol/cli/releases
+
+## Installing or Updating Oasis CLI on macOS
+
+This guide covers the installation or update process for the Oasis CLI on
+macOS (compatible with Apple Silicon like M1/M2/M3). It assumes you have
+Terminal access and basic command-line knowledge.
+
+### Prerequisites
+
+- macOS (tested on Apple Silicon).
+- To avoid modifying system directories, we'll install the binary in a
+  user-specific path (`~/.local/bin`). Ensure `~/.local/bin` is in your `PATH`.
+  If not, add it by editing your shell configuration file (e.g., `~/.zshrc`
+  for zsh):
+
+  ```bash
```

Comment:
These commands are not bash-specific, so `shell` should work just fine. Same below.

````suggestion
  ```shell
````